org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/StatusWidget.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xtext.ui.wizard.project;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.IMessageProvider;
+import org.eclipse.jface.resource.JFaceResources;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.xtext.xbase.lib.Procedures.Procedure0;
+
+/**
+ * @author dhuebner - Initial contribution and API
+ */
+public class StatusWidget extends Composite {
+
+	private Link link;
+
+	private Label imageLabel;
+
+	private Procedure0 quickFix = () -> {};
+
+	private int severity = IMessageProvider.NONE;
+	
+	private static final Pattern ANCHOR_PATTERN = Pattern.compile("<a>(.*)</a>");
+
+	public StatusWidget(Composite parent, int style) {
+		super(parent, style);
+		createControls();
+	}
+
+	protected void createControls() {
+		setLayout(new GridLayout(2, false));
+
+		imageLabel = new Label(this, SWT.NONE);
+		imageLabel.setText("   ");
+		GridData imgLabelLayoutData = new GridData(GridData.VERTICAL_ALIGN_BEGINNING);
+		imgLabelLayoutData.widthHint = 20;
```

Comment:
PS image below. Changed the value of `widthHint` from 15 to 20 pixels. There were cases where I found that the image was getting cropped. Changing it to 20 pixes resolves that issue, also the additional pixels improves the user experience.

![image](https://user-images.githubusercontent.com/3328139/77846757-e3150600-71d5-11ea-82c0-085a2d03090a.png)